samples/sensor/bme280/sample.yaml

```diff
@@ -17,19 +17,18 @@ tests:
   sample.sensor.bme280.spi:
     harness: console
     tags: sensors
-    depends_on:
```

Comment:
this should be looking for a bme280 instead, e.g. dt_compat_enable("bosch,bme280"); and then using the appropriate DT macro to find the first bme280 on a SPI bus imo, setting platform_allow to ruuvitag here looks wrong.

Reply:
this scenario is not buiulding for anything right now, it was wrong already, adding `ruuvi_ruuvitag` which has the sensor on SPI finally get it to build:

```
i9:bme280(a64.ld_fix): west twister -s sample.sensor.bme280.spi -v --all
Renaming output directory to /home/nashif/zephyrproject/zephyr/samples/sensor/bme280/twister-out.1
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-2153-g31f3bf28a572
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting all possible platforms per test case
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/samples/sensor/bme280/twister-out/testplan.json
INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue

INFO    - 1 test scenarios (822 test instances) selected, 822 configurations skipped (822 by static filter, 0 at runtime).
INFO    - 0 of 822 test configurations passed (0.00%), 0 failed, 0 errored, 822 skipped with 0 warnings in 5.37 seconds
INFO    - In total 0 test cases were executed, 822 skipped on 822 out of total 823 platforms (99.88%)
INFO    - 0 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/samples/sensor/bme280/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/samples/sensor/bme280/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/samples/sensor/bme280/twister-out/twister_report.xml...
INFO    - Run completed
```

with this change....

```
820/820 ruuvi_ruuvitag            samples/sensor/bme280/sample.sensor.bme280.spi     PASSED (build)
```

Reply:
using filter means I have to go through all boards and see which one does have this sensor, wasting lots of resources. This is just a sample, if we know which platform has this configuration and sensor, we should not use filter, instead just allow that platform.

Reply:
Is this because of the whole dt requires a CMake run and build target built before seeing the full devicetree thing I ran into awhile back? Ugh

Reply:
yes